extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/ByteChunkReader.java

```diff
@@ -0,0 +1,204 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+// ****** AUTO-GENERATED CLASS - DO NOT EDIT MANUALLY
+// ****** Edit CharChunkReader and run "./gradlew replicateBarrageUtils" to regenerate
+//
+// @formatter:off
+package io.deephaven.extensions.barrage.chunk;
+
+import io.deephaven.base.verify.Assert;
+import io.deephaven.chunk.WritableByteChunk;
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.WritableLongChunk;
+import io.deephaven.chunk.WritableObjectChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.extensions.barrage.util.StreamReaderOptions;
+import io.deephaven.util.datastructures.LongSizedDataStructure;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.PrimitiveIterator;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+
+import static io.deephaven.util.QueryConstants.NULL_BYTE;
+
+public class ByteChunkReader implements ChunkReader {
+    private static final String DEBUG_NAME = "ByteChunkReader";
+    private final StreamReaderOptions options;
+    private final ByteConversion conversion;
+
+    @FunctionalInterface
+    public interface ByteConversion {
+        byte apply(byte in);
+
+        ByteConversion IDENTITY = (byte a) -> a;
+    }
+
+    public ByteChunkReader(StreamReaderOptions options) {
+        this(options, ByteConversion.IDENTITY);
+    }
+
+    public ByteChunkReader(StreamReaderOptions options, ByteConversion conversion) {
+        this.options = options;
+        this.conversion = conversion;
+    }
+
+    public <T> ChunkReader transform(Function<Byte, T> transform) {
```

Comment:
We may want to replace these `Function<BoxedPrimitive, T>` interfaces with some replicated version to avoid unnecessary boxing of values that can never be null.

Reply:
In the interest of moving this patch along, I'm going to punt on this, the updated version is no more wrong than it previously was.

Reply:
Although, we may want the transformers to "transform" null values to give better control to the future-feature of custom formatters. If this were the case then the boxing would be necessary.

Reply:
I _believe_ that these already have the DH null values as primitives, so a `null` input is impossible at this time. `T` could certainly be null though for an output, depending on what kind of chunk is going to be written to (not controlled by this code).

Reply:
I simply notice that we do not call the transformer if the value is the null value. As in, a custom transformer can't react to any null values. I see your point that we could use the deephaven null value _and_ that using primitives will make it very clear that null is being represented in a non-boxed way. You're probably right that we should avoid boxing here.